lti_consumer/plugin/views.py

```diff
@@ -118,53 +128,204 @@ def public_keyset_endpoint(request, usage_id=None):
         response['Content-Disposition'] = 'attachment; filename=keyset.json'
         return response
     except (LtiError, InvalidKeyError, ObjectDoesNotExist) as exc:
-        log.info("Error while retrieving keyset for usage_id %r: %s", usage_id, exc)
+        log.info(
+            "Error while retrieving keyset for usage_id (%r) or lit_config_id (%s): %s",
+            usage_id,
+            lti_config_id,
+            exc
+        )
         raise Http404 from exc
 
 
 @require_http_methods(["GET", "POST"])
-def launch_gate_endpoint(request, suffix=None):
+def launch_gate_endpoint(request, suffix=None):  # pylint: disable=unused-argument
     """
     Gate endpoint that triggers LTI launch endpoint XBlock handler
 
-    This is basically a passthrough function that uses the
-    OIDC response parameter `login_hint` to locate the block
-    and run the proper handler.
+    This uses the location key from the "login_hint" query parameter
+    to identify the LtiConfiguration and its consumer to generate the
+    LTI 1.3 Launch Form.
     """
+    usage_id = request.GET.get('login_hint')
```

Comment:
@giovannicimolin I tried implementing using either the location OR the config uuid. But ran into issues down the line as the [launch context information](https://github.com/openedx/xblock-lti-consumer/blob/master/lti_consumer/lti_xblock.py#L1191-L1216) is all dependent on the course, which is being obtained from the UsageKey. So, until we figure out how to establish the context (user_id, user_role, context_title, content_types...etc.,) using the `config_id` value in `login_hint` might not be possible.

Reply:
@tecoholic Good point! 

This means we'll need to modify either the `login_hint` or the `lti_message_hint` to pass the id along with the appropriate context. Otherwise, we won't be able to make LTI 1.3 configurations really independent of blocks.

Let me think about a solution today - I'll try to comment back before my EOD.